File: AS-62/gritsuk/lab2/app/src/main/java/com/example/labjson/MainActivity.java
package com.example.labjson;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.ProgressBar;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;


import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.IOException;
import java.io.InputStream;
import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private static final String JSON_FILE_NAME = "item.json";
    ListView listView;

    EditText editTextUserCount;
    Button buttonLoadUsers;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        listView = findViewById(R.id.listView);
        editTextUserCount = findViewById(R.id.editTextUserCount);
        buttonLoadUsers = findViewById(R.id.buttonLoadUsers);
        buttonLoadUsers.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // Получите введенное пользователем число
                String countText = editTextUserCount.getText().toString();
                if (!countText.isEmpty()) {
                    int userCount = Integer.parseInt(countText);
                    loadJSONFromFile(JSON_FILE_NAME, userCount);
                } else {
                    Toast.makeText(getApplicationContext(), "Введите число пользователей", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }

    private void loadJSONFromFile(String fileName, int userCount) {
        final ProgressBar progressBar = (ProgressBar) findViewById(R.id.progressBar);
        progressBar.setVisibility(ListView.VISIBLE);
        String jsonString = null;

        try {
            InputStream inputStream = getAssets().open(fileName);
            int size = inputStream.available();
            byte[] buffer = new byte[size];
            inputStream.read(buffer);
            inputStream.close();
            jsonString = new String(buffer, "UTF-8");
            // Обработка JSON-данных и ограничение количества пользователей
            JSONArray jsonArray = new JSONArray(jsonString);
            ArrayList<JSONObject> listItems = getArrayListFromJSONArray(jsonArray);

            if (userCount < listItems.size()) {
                listItems.subList(userCount, listItems.size()).clear();
            }

            ListAdapter adapter = new ListViewAdapter(getApplicationContext(), R.layout.raw, R.id.textViewName, listItems);
            listView.setAdapter(adapter);

            progressBar.setVisibility(View.INVISIBLE);
        } catch (IOException | JSONException e) {
            e.printStackTrace();
            progressBar.setVisibility(View.INVISIBLE);
            Toast.makeText(getApplicationContext(), "Ошибка чтения файла JSON", Toast.LENGTH_SHORT).show();
        }
    }



    private ArrayList< JSONObject> getArrayListFromJSONArray(JSONArray jsonArray){
        ArrayList< JSONObject> aList = new ArrayList< JSONObject>();
        try {
            if(jsonArray!= null){
                for(int i = 0; i< jsonArray.length();i++){
                    aList.add(jsonArray.getJSONObject(i));
                }
            }
        }catch (JSONException js){
            js.printStackTrace();
        }
        return aList;
    }



}

File: AS-62/kulick/lab2/sources/app/src/main/java/com/example/lab2/MainActivity.kt
package com.example.lab2

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import android.view.View


class MainActivity : AppCompatActivity() {

    private lateinit var tvInfo: TextView //переменные для полей ввода и кнопки
    private lateinit var etInput: EditText
    private lateinit var bControl: Button
    var guess = 0 //загаданное число
    var gameFinished = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tvInfo = findViewById(R.id.textView);
        etInput = findViewById(R.id.editTextTextPersonName);
        bControl = findViewById(R.id.button);
        guess = (Math.random()*100).toInt(); //рандом числа
        gameFinished = false; //переменная окончания игры


    }
    fun onClick(v: View?){
        if (!gameFinished) { //если игра не закончена

                if (etInput.text.toString().isEmpty()) { //если строка пустая а кнопка нажата
                    tvInfo.text = resources.getString(R.string.error); //ошибка
                    return

                }
            var inp =
                Integer.parseInt(etInput.text.toString()); //считываем ввод в строку и переводим в число
            if (!(inp in 0..100)) {
                tvInfo.text = resources.getString(R.string.error);
            } else {
                if (inp > guess) //если бошльше загаданного
                    tvInfo.text = resources.getString(R.string.ahead);
                if (inp < guess) //если меньше загаданного
                    tvInfo.text = resources.getString(R.string.behind);
                if (inp == guess) //если угадал
                {
                    tvInfo.text = resources.getString(R.string.hit);
                    bControl.text = resources.getString(R.string.play_more);
                    gameFinished = true; //игра окончена
                }

            }
        }
        else //если игра окончена
        {
            guess = ((Math.random()*100).toInt()); //рандом нового числа
            bControl.text = resources.getString(R.string.input_value);
            tvInfo.text = resources.getString(R.string.try_to_guess);
            gameFinished = false;
        }
        etInput.text; //вывод исходного текста в строку ввода
    }
}

